### ISSUE TITLE
cdll.LoadLibrary allows None as an argument

### ISSUE THREAD

**Issue by 47122e6f-3270-44bd-b9d5-db5f1e7afcf8** [python/cpython#78773]
BPO | [34592](https://bugs.python.org/issue34592)
--- | :---
Nosy | @gpshead, @vstinner, @superbobry, @mhsmith

<sup>*Note: these values reflect the state of the issue at the time it was migrated and might not reflect the current state.*</sup>

<details><summary>Show more details</summary><p>

GitHub fields:
```python
assignee = None
closed_at = None
created_at = <Date 2018-09-05.21:35:41.405>
labels = ['3.11', 'invalid', 'docs']
title = 'cdll.LoadLibrary allows None as an argument'
updated_at = <Date 2021-11-19.04:06:58.621>
user = 'https://github.com/superbobry'
```

bugs.python.org fields:
```python
activity = <Date 2021-11-19.04:06:58.621>
actor = 'gregory.p.smith'
assignee = 'docs@python'
closed = False
closed_date = None
closer = None
components = ['Documentation']
creation = <Date 2018-09-05.21:35:41.405>
creator = 'superbobry'
dependencies = []
files = []
hgrepos = []
issue_num = 34592
keywords = []
message_count = 5.0
messages = ['324654', '361734', '361735', '361807', '406570']
nosy_count = 5.0
nosy_names = ['gregory.p.smith', 'vstinner', 'docs@python', 'superbobry', 'Malcolm Smith']
pr_nums = []
priority = 'normal'
resolution = 'not a bug'
stage = 'needs patch'
status = 'open'
superseder = None
type = None
url = 'https://bugs.python.org/issue34592'
versions = ['Python 3.11']
```

</p></details>

### Linked PRs
* gh-145313
* gh-145674

**Comment by encukou** [comment]
> I'm assuming it likely does on everything POSIXy.

Indeed. [POSIX says](https://pubs.opengroup.org/onlinepubs/9799919799/functions/dlopen.html):

> If *file* is a null pointer, `dlopen()` shall return a global symbol table handle for the currently running process image. This symbol table handle shall provide access to the symbols from an ordered set of executable object files consisting of the original program image file, any executable object files loaded at program start-up as specified by that process file (for example, shared libraries), and the set of executable object files loaded using `dlopen()` operations with the `RTLD_GLOBAL` flag. As the latter set of executable object files can change during execution, the set of symbols made available by this symbol table handle can also change dynamically.
